### PR TITLE
[Spike] Check for children more efficiently during clean-up

### PR DESCRIPTION
**Pull request by seiffert**
**Do not merge**, this PR needs to be cleaned up and then committed to `hashicorp/vault`.

### CODE REVIEW

File: physical/dynamodb.go [hunk 1]
```diff
@@ -378,6 +378,36 @@ func (d *DynamoDBBackend) List(prefix string) ([]string, error) {
...
+		ConsistentRead: aws.Bool(true),
+		KeyConditions: map[string]*dynamodb.Condition{
+			"Path": {
+				ComparisonOperator: aws.String("EQ"),
+				AttributeValueList: []*dynamodb.AttributeValue{{
+					S: aws.String(prefix),
+				}},
+			},
+		},
+		Limit: aws.Int64(2),
```
Comment (seiffert) [inline physical/dynamodb.go:398]: This was @TobiasBales's idea! ❤️

**Comment by seiffert** [comment]
Binaries built from this branch: [linux_amd64.zip](https://github.com/Jimdo/vault/files/994511/linux_amd64.zip)

File: physical/dynamodb.go (hunk 1 above)
Comment (TobiasBales) [reply physical/dynamodb.go:398]: I think you should document why the limit is 2 and not 1 (in a comment I'd guess)

Comment (seiffert) [reply physical/dynamodb.go:398]: yes of course! This is just a spike to validate the idea 😉

**Comment by seiffert** [comment]
this is nowadays part of Vault: https://github.com/hashicorp/vault/pull/2722

**Pull request closed without merging**